### PR TITLE
Initial packaging work

### PR DESCRIPTION
**Pull request by JayH5**
- Add download script
- Sideloader .deploy.yaml
- Example config files
- Upstart script
- Update README

### CODE REVIEW

**Comment by JayH5** [comment]
@jerith: Ready for review.

I've done things slightly differently in the build script:
- The SHA256SUMS file is downloaded from Hashicorp rather than there being a sha256sums.txt file in the repo.
- The gpg signature file for the SHA256SUMS file is also downloaded and used to check the authenticity of the file. This requires that the Hashicorp public key is in our default GPG keyring.

All this signature checking stuff is probably overkill or good practice. I'm not sure which :/ If we do it this way I need to add the Hashicorp public key to the Sideloader servers using Puppet.

**Comment by jerith** [comment]
:+1:

**Comment by JayH5** [comment]
@jerith: Please review two small changes after testing on Sideloader.

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+# Move contents to correct location in package
+# Hack sideloader to not build things in /var/praekelt - install files straight into the package directory
+PACKAGE_DIR=$WORKSPACE/package
+mkdir -p $PACKAGE_DIR
+mkdir -p $PACKAGE_DIR$BIN_DIR
+# Move the binary in to place
+mv ./$PROJECT $PACKAGE_DIR$BIN_DIR
+# Copy the package files
+cp -r $(pwd)/$REPO/package/* $PACKAGE_DIR
```
Comment (jerith) [inline build.sh]: You probably want `cp -r "$(pwd)/$REPO/package"/* "$PACKAGE_DIR"` to handle potential spaces in paths. The `*` needs to be unquoted, but the variables and subshell commands should be quoted.

**Comment by jerith** [comment]
Similar to my comment above, we should add appropriate quoting to the other places we're using paths from envvars and such.

**Comment by jerith** [comment]
:+1:

**Pull request merged**